pkg/operator/controller/certificate/controller.go

```diff
@@ -77,119 +59,56 @@ type reconciler struct {
+				APIVersion: "apps/v1",
+				Kind:       "Deployment",
+				Name:       deployment.Name,
+				UID:        deployment.UID,
+				Controller: &trueVar,
+			}
+			err = r.ensureDefaultCertificateForIngress(ca, deployment.Namespace, deploymentRef, ingress)
+			if err != nil {
+				errs = append(errs, fmt.Errorf("failed to ensure default cert for %s: %v", ingress.Name, err))
+			}
 		}
-		return nil, err
-	}
-	return secret, nil
-}
-
-// generateRouterCA generates and returns a CA certificate and key.
-func generateRouterCA() ([]byte, []byte, error) {
-	signerName := fmt.Sprintf("%s@%d", "cluster-ingress-operator", time.Now().Unix())
-
-	privateKey, err := rsa.GenerateKey(rand.Reader, 2048)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to generate key: %v", err)
-	}
-
-	root := &x509.Certificate{
-		Subject: pkix.Name{CommonName: signerName},
-
-		SignatureAlgorithm: x509.SHA256WithRSA,
-
-		NotBefore:    time.Now().Add(-1 * time.Second),
-		NotAfter:     time.Now().Add(2 * 365 * 24 * time.Hour),
-		SerialNumber: big.NewInt(1),
-
-		KeyUsage:              x509.KeyUsageKeyEncipherment | x509.KeyUsageDigitalSignature | x509.KeyUsageCertSign,
-		BasicConstraintsValid: true,
-
-		IsCA: true,
-
-		// Don't allow the CA to be used to make another CA.
-		MaxPathLen:     0,
-		MaxPathLenZero: true,
-	}
-
-	derBytes, err := x509.CreateCertificate(rand.Reader, root, root, &privateKey.PublicKey, privateKey)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to create certificate: %v", err)
-	}
-
-	certs, err := x509.ParseCertificates(derBytes)
-	if err != nil {
-		return nil, nil, fmt.Errorf("failed to parse certificate: %v", err)
-	}
-
-	if len(certs) != 1 {
-		return nil, nil, fmt.Errorf("expected a single certificate")
-	}
-
-	certBytes := pem.EncodeToMemory(&pem.Block{
-		Type:  "CERTIFICATE",
-		Bytes: certs[0].Raw,
-	})
-
-	keyBytes := pem.EncodeToMemory(&pem.Block{
-		Type:  "RSA PRIVATE KEY",
-		Bytes: x509.MarshalPKCS1PrivateKey(privateKey),
-	})
-
-	return certBytes, keyBytes, nil
-}
-
-// desiredRouterCASecret returns the desired router CA secret.
-func desiredRouterCASecret(namespace string) (*corev1.Secret, error) {
-	certBytes, keyBytes, err := generateRouterCA()
-	if err != nil {
-		return nil, fmt.Errorf("failed to generate certificate: %v", err)
-	}
-
-	name := CASecretName(namespace)
-	secret := &corev1.Secret{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      name.Name,
-			Namespace: name.Namespace,
-		},
-		Data: map[string][]byte{
-			"tls.crt": certBytes,
-			"tls.key": keyBytes,
-		},
-		Type: corev1.SecretTypeTLS,
 	}
-	return secret, nil
+	return reconcile.Result{}, utilerrors.NewAggregate(errs)
 }
 
-// createRouterCASecret creates the router CA secret.
-func (r *reconciler) createRouterCASecret(secret *corev1.Secret) error {
-	if err := r.client.Create(context.TODO(), secret); err != nil {
-		return err
+func routerDeploymentName(ci *ingressv1alpha1.ClusterIngress) types.NamespacedName {
```

Comment:
I think you suggested moving [Resource]Name to controller/names.go or something on slack. Do we want to do that as a separate pr or is it dropped for some other reason?

Reply:
Started the cleanup of name handling in https://github.com/openshift/cluster-ingress-operator/pull/142; too much work to do here because of import cycles